react/CipherIcon/index.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import AppIcon from '../AppIcon'
+import Icon from '../Icon'
+import { withClient } from 'cozy-client'
+import { AppDoctype } from '../proptypes'
+
+class CipherIcon extends React.PureComponent {
+  constructor(props) {
+    super(props)
+    this.fetchIcon = this.fetchIcon.bind(this)
+  }
+
+  fetchIcon() {
+    const { client, konnector } = this.props
+
+    return client.stackClient.getIconURL({
+      type: 'konnector',
+      slug: konnector.slug || konnector
+    })
+  }
```

Comment:
In https://github.com/cozy/cozy-ui/pull/1152 I will work so that user provided fetchIcon gets props automatically. This will remove the need to have to bind the fetchIcon in the constructor and will enable fetchIcon to be completely extracted.

```
// In another file
const fetchKonnectorIcon = ({ client, konnector }) => 
   client.stackClient.getIconURL({
      type: 'konnector',
      slug: konnector.slug || konnector
    })

const CipherIcon = React.memo(({ className }) => (
      <div className={cx('u-dib u-pos-relative', className)} {...rest}>
        <div className="u-pos-absolute u-right-0 u-bottom-0">
          <AppIcon fetchIcon={fetchKonnectorIcon} className="u-w-1 u-h-1" />
        </div>
        <Icon icon="keychain" size={32} />
      </div>
    ))
```

Reply:
:+1: